emukit/bayesian_optimization/local_penalization_calculator.py

```diff
@@ -0,0 +1,93 @@
+import numpy as np
+import scipy.optimize
+
+from emukit.core import ParameterSpace
+from emukit.core.acquisition import Acquisition
+from emukit.core.interfaces import IDifferentiable
+from emukit.core.loop import CandidatePointCalculator, LoopState
+from emukit.bayesian_optimization.acquisitions.local_penalization import LocalPenalization
+from emukit.core.optimization import AcquisitionOptimizer
+from emukit.experimental_design.model_free.random_design import RandomDesign
+
+N_SAMPLES = 500  # Number of samples to use when estimating Lipschitz constant
+MAX_ITER = 200  # Maximum number of iterations for optimizer when estimating Lipschitz constant
+
+
+class LocalPenalizationPointCalculator(CandidatePointCalculator):
+    """
+    Candidate point calculator that computes a batch using local penalization from:
+
+    `Batch Bayesian Optimization via Local Penalization. Javier González, Zhenwen Dai, Philipp Hennig, Neil D. Lawrence
+    <https://arxiv.org/abs/1505.08052>`_
+    """
+
+    def __init__(self, acquisition: Acquisition, acquisition_optimizer: AcquisitionOptimizer, model: IDifferentiable,
+                 parameter_space: ParameterSpace, batch_size: int):
+        """
+        :param acquisition: Base acquisition function to use without any penalization applied, this acquisition should
```

Comment:
The type for this guy should be explicitly LocalPenalizationAcquisition, as it will surely fail for anything else

Reply:
oh, right, thanks for explanation